### DISCUSSION TITLE
Interpreting ccache debugging info

### DISCUSSION THREAD

**Discussion by ftrofin** [ccache/ccache#827] [Support]
Hello all,

I'm trying to figure out why I'm getting less cache hits than I'd like to. The context is this:
- ccache 4.2
- Single repository (so the builds happen in the same base dir)
- Direct mode is on
- cache size is more than reasonable (80GB total, used only 40GB)
- The other settings:
```
(default) direct_mode = true
(default) disable = false
(default) extra_files_to_hash = 
(default) file_clone = false
(default) hard_link = false
(default) hash_dir = true
(default) ignore_headers_in_manifest = 
(default) ignore_options = 
(default) inode_cache = false
(default) keep_comments_cpp = false
(default) limit_multiple = 0.8
(default) log_file = 
(default) max_files = 0
(~/.ccache/ccache.conf) max_size = 80.0G
(default) path = 
(default) pch_external_checksum = false
(default) prefix_command = 
(default) prefix_command_cpp = 
(default) read_only = false
(default) read_only_direct = false
(default) recache = false
(default) run_second_cpp = true
(default) sloppiness = 
(default) stats = true
(default) temporary_dir = /Volumes/DATA/CCACHE/tmp
(default) umask = 
```

Diffing the *.cache-input-text files (between the builds) only renders one or two strings different,  like `### include.c110rdevfg...` (some hash). 

Diffing the log files for the same object file shows only the timestamps being different and the location in the cache (`like /Volumes/DATA/CCACHE/f/5/7c9uckcbfpkj8630llhcbdrrbrumdsiR` being different. The command lines are identical.

What does this mean? It seems to be something related to some `include` (some header that changed)? How can I track down what's causing the cache misses?

Thanks a lot for your help!

**Comment by jrosdahl** [comment]
That's impossible to say without more information. Can you provide `.ccache-input-*` and `.ccache-log` files for two compilations where the second should be identical to the first but resulted in a cache miss?
